CMakeLists.txt

```diff
@@ -3,10 +3,22 @@ cmake_minimum_required(VERSION 3.10)
 set(LJMD_VERSION 1.0 CACHE STRING "Floating point number for the current version")
 project(ljmd VERSION ${LJMD_VERSION} LANGUAGES C)
 
-add_executable(ljmd.x main.c src/utilities.c src/input_output.c src/engine.c )
+add_executable(ljmd.x
+	main.c
+)
 
-target_include_directories(ljmd.x PRIVATE "include")
+add_library(ljmd_rt SHARED
+    src/engine.c
+    src/input_output.c
+    src/utilities.c
+)
+
+target_include_directories(ljmd.x PRIVATE include)
```

Comment:
in these multiple target_* lines, I think we should separate those regarding the lib and those regarding the exe
so, for example: first do all commands for the lib, then all commands for the exe